app/models/match.rb

```diff
@@ -41,6 +41,23 @@ def self.follow_up_status(follow_up_status)
     result
   end
 
+  def self.create_match_for_contribution!(contribution, current_user)
+    match_params = if contribution.ask?
+                      { receiver: contribution, provider: create_offer_for_ask!(contribution, current_user) }
+                    elsif contribution.offer? # TODO: check if community resource type when it's added
+                      { receiver: create_ask_for_offer!(contribution, current_user), provider: contribution }
+                    end
+    Match.create!(match_params)
+  end
+
+  def self.create_offer_for_ask!(ask, current_user)
+    Offer.create!(person: current_user.person, service_area: ask.service_area)
+  end
+
+  def self.create_ask_for_offer!(offer, current_user)
+    Ask.create!(person: current_user.person, service_area: offer.service_area)
+  end
```

Comment:
@maebeale , is this sufficient info on the auto-created Ask|Offer? Should we add the category as well?

Reply:
ooo. yes, we *should* add Category to the autogenerated Ask/Offer